tsuikm/usdc-dashboard
# Masterminter role page

# Contributors
@constanceshi 

# Relevant issue
brought up in demo

# Summary of change
- Added a roles/masterminter route 
- Added input to check if an address is a minter
- Added functionality to remove minters, increase/change their allowance, or configure new minters with a specified allowance
- It currently has the same design as the old blacklister control page, but I can also change it to something new!
- styled the page using custom inputs and buttons
- added line heights for other role pages so that on smaller screen size the text in the title will not overlap itself
![image](https://user-images.githubusercontent.com/56846508/100027059-d8c8fd00-2da0-11eb-8725-346231fc13b7.png)
![image](https://user-images.githubusercontent.com/56846508/100027071-e088a180-2da0-11eb-93f4-ca4eb46c4a27.png)


# Testing plan
- Created masterminter.spec.js to test display and functionality
- Tested via a locally hosted blockchain:
    - Instructions:
        - start the locally hosted blockchain following the instructions in the wiki except export the usdc contract address as the contract address under the outputs of truffle migrate and the one that says Deploying 'FiatTokenV2'
        - go to the roles/owner page, import the owner account into your metamask (first account in the console of the ganache console thing), and connect and switch to this account
        - make this account or an account of your choice the masterminter
        - navigate to roles/masterminter and switch your account on metamask to whichever account you made the masterminter
        - choose an address to check the status of and eventually make a minter
        - check the status (should not be a minter)
        - configure the minter with an allowance of your choice (remember, this allowance will be divided by 6 decimal places [1 million] in usdc)
        - the page will take a few seconds to update but should change to display this address is a minter with allowance whatever you entered
        - you can navigate to /mint route and try to mint if you import this minter account into your metamask and switch to it
        - choose an account to mint to and choose an amount (must be less than the minter's allowance divided by 1 million)
        - mint and approve the transaction and wait
        - to see the increase in balance of the account you minted to, import the account and add a token with the contract address of the one you exported as the usdc contract address
        - you can go back to the roles/masterminter route to increase/change the allowance or remove the minter and play around with that, just remember to connect to the corresponding accounts on metamask depending on what you are trying to do at any given point in time
